restapi/src/main/java/io/stargate/web/docsapi/service/util/DocsApiUtils.java

```diff
@@ -0,0 +1,231 @@
+          .collect(Collectors.joining(","));
+    } else {
+      return convertSingleArrayPath(path);
+    }
+  }
+
+  private static String convertSingleArrayPath(String path) {
+    // check if we have array path
+    if (ARRAY_PATH_PATTERN.matcher(path).matches()) {
+      String innerPath = path.substring(1, path.length() - 1);
+      // if it's wildcard keep as it is
+      if (!Objects.equals(innerPath, DocumentDB.GLOB_VALUE)) {
+        // otherwise try to parse int
+        try {
+          // this can fail, thus wrap in the try
+          int idx = Integer.parseInt(innerPath);
+          if (idx > DocumentDB.MAX_ARRAY_LENGTH - 1) {
+            String msg =
+                String.format("Max array length of %s exceeded.", DocumentDB.MAX_ARRAY_LENGTH);
+            throw new ErrorCodeRuntimeException(
+                ErrorCode.DOCS_API_GENERAL_ARRAY_LENGTH_EXCEEDED, msg);
+          }
+          return "[" + leftPadTo6(innerPath) + "]";
+        } catch (NumberFormatException e) {
+          String msg = String.format("Array path %s is not valid.", path);
+          throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_SEARCH_ARRAY_PATH_INVALID, msg);
+        }
+      }
+    }
+    return path;
+  }
+
+  public static String leftPadTo6(String value) {
+    return StringUtils.leftPad(value, 6, '0');
+  }
+
+  /**
+   * Returns the collection of paths that are represented by each JsonNode in the #fieldsJson.
+   *
+   * <p>For each field a list of strings representing the path is returned. Note that for each path
+   * member, {@link #convertArrayPath(String)} will be called.
+   *
+   * @param fieldsJson array json node
+   * @return collection of paths representing all fields
+   */
+  public static Collection<List<String>> convertFieldsToPaths(JsonNode fieldsJson) {
+    if (!fieldsJson.isArray()) {
+      String msg = String.format("`fields` must be a JSON array, found %s", fieldsJson);
+      throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_GENERAL_FIELDS_INVALID, msg);
+    }
+
+    Collection<List<String>> results = new ArrayList<>();
+    for (JsonNode value : fieldsJson) {
+      if (!value.isTextual()) {
+        String msg = String.format("Each field must be a string, found %s", value);
+        throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_GENERAL_FIELDS_INVALID, msg);
+      }
+      String fieldValue = value.asText();
+      List<String> fieldPath =
+          Arrays.stream(fieldValue.split("\\."))
+              .map(DocsApiUtils::convertArrayPath)
+              .collect(Collectors.toList());
+
+      results.add(fieldPath);
+    }
+
+    return results;
+  }
+
+  /**
+   * Simple utility to return a string value from doc api {@link Row}.
+   *
+   * @param row Row
+   * @return String or null if a row column {@value QueryConstants#STRING_VALUE_COLUMN_NAME) is null.
+   */
+  public static String getStringFromRow(Row row) {
+    return row.isNull(QueryConstants.STRING_VALUE_COLUMN_NAME)
```

Comment:
It might be better to avoid the double hash lookup e.g.
```
int index = row.firstIndexOf(QueryConstants.STRING_VALUE_COLUMN_NAME);
return row.isNull(index) ? null : row.getString(index);
```